filament/backend/test/BackendTest.cpp

```diff
@@ -114,31 +115,38 @@ PipelineState BackendTest::getColorWritePipelineState() {
     return result;
 }
 
-void BackendTest::fullViewport(RenderPassParams& params) {
-    fullViewport(params.viewport);
+filament::backend::Viewport BackendTest::getFullViewport() const {
+   const NativeView& view = getNativeView();
+   return Viewport {
+       .left = 0,
+       .bottom = 0,
+       .width = static_cast<uint32_t>(view.width),
+       .height = static_cast<uint32_t>(view.height)
+   };
 }
 
-void BackendTest::fullViewport(Viewport& viewport) {
-    const NativeView& view = getNativeView();
-    viewport.left = 0;
-    viewport.bottom = 0;
-    viewport.width = view.width;
-    viewport.height = view.height;
+filament::backend::RenderPassParams BackendTest::getClearColorRenderPass(
+        std::optional<float4> color) {
+    RenderPassParams params = {};
+    params.flags.clear = TargetBufferFlags::COLOR;
+    params.flags.discardStart = TargetBufferFlags::ALL;
+    params.flags.discardEnd = TargetBufferFlags::NONE;
+    params.clearColor = color.value_or(float4(0, 1, 1, 1));
```

Comment:
Just curious, is there an advantage to using a `std::optional` here as opposed to just a default argument? It might be better to use a default argument so the default value is clearer to the users of the API.

Reply:
It's a rust pattern leaking in. The downside to a default value is that now your header contains an implementation detail. Which is pretty minor, you're correct that just using a default value is the right approach here.